dev/com.ibm.ws.el.3.0_fat/test-applications/TestVarargsMatching.war/src/com/ibm/ws/el30/fat/varargstest/EL30VarargsMethodMatchingServlet.java

```diff
@@ -69,42 +80,74 @@ public void testSingleString() throws Exception {
 
     }
 
     @Test
     @SkipForRepeat(SkipForRepeat.EE9_FEATURES)
     public void testString_String() throws Exception {
 
-        getMethodExpression("testBean.testMethod('string1','string2')", "(String param1, String param2)");
+        getMethodExpression("testBean.testMethod('string1', 'string2')", "(String param1, String param2)");
 
     }
 
     @Test
     @SkipForRepeat(SkipForRepeat.EE9_FEATURES)
     public void testEnum_VarargsEnum() throws Exception {
         
-        getMethodExpression("testBean.testMethod(testBean.enum1,testBean.enum1,testBean.enum1)", "(IEnum enum1, IEnum... enum2)");
+        getMethodExpression("testBean.testMethod(enum1, enum1, enum1)", "(IEnum enum1, IEnum... enum2)");
 
     }
 
     @Test
     @SkipForRepeat(SkipForRepeat.EE9_FEATURES)
-    public void testString_VarargsEnum() throws Exception {
+    public void testString_VarargsMultipleEnum() throws Exception {
         
-        getMethodExpression("testBean.testMethod('string1',testBean.enum1)", "(String param1, IEnum... param2)");
+        getMethodExpression("testBean.testMethod('string1', enum1, enum1)", "(String param1, IEnum... param2)");
 
     }
 
     @Test
     @SkipForRepeat(SkipForRepeat.EE9_FEATURES)
-    public void testVarargsInt() throws Exception {
+    public void selectMethodWithNoVarargs() throws Exception {
         
-        getMethodExpression("testBean.testMethod(testBean.number)", "(int... param1)");
+        getMethodExpression("testBean.chirp(falcon)", "chirp(Bird bird1)");
 
     }
 
+    @Test
+    @SkipForRepeat(SkipForRepeat.EE9_FEATURES)
+    public void testString_VarargsBird() throws Exception {
+        
+        getMethodExpression("testBean.chirp('string1', bird, bird)", "chirp(String string1, Bird... bird2)");
+
+    }
+
+    // Limitions of the varags selection are below -- tests currently fail 
+    // See Mark's July 29th comment here: https://bz.apache.org/bugzilla/show_bug.cgi?id=65358
+
+    // No varargs methods are always prefered over varargs 
+    // @Test
+    // @SkipForRepeat(SkipForRepeat.EE9_FEATURES)
+    // public void testVarargsInt() throws Exception {
+    //     getMethodExpression("testBean.testMethod(number)", "(int... param1)");
+    // }
+
+    // Failure related to coercion, I think? 
+    // @Test
+    // @SkipForRepeat(SkipForRepeat.EE9_FEATURES)
+    // public void testString_VarargsEnum() throws Exception {
+    //     getMethodExpression("testBean.testMethod('string1', enum1)", "(String param1, IEnum... param2)");
+    // }
+
+    // Not sure if related to varargs, but it fails nonetheless. -- MethodNotFoundException "Unable to find unambiguous method"
+    // @Test
+    // @SkipForRepeat(SkipForRepeat.EE9_FEATURES)
+    // public void testString_VarargsFalcon() throws Exception { 
+    //     getMethodExpression("testBean.chirp('string1', falcon, falcon)", "chirp(String string1, Bird... bird2)");
+    // }
+
```

Comment:
This is an odd one considering the MethodNotFoundException? I can follow up with Tomcat, if you prefer.   But I clumped it under "EL limitations" anyway.